### PR TITLE
Add PMD checks

### PR DESCRIPTION
**Pull request by rkuris**
Cherry picked the PMD set from a larger list, and corrected a few warnings

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -44,3 +45,10 @@ dependencies {
     testCompile group: 'com.google.collections', name: 'google-collections', version: '1.0'
     testCompile group: 'junit', name: 'junit', version: '4.11'
 }
+
+pmd {
+  targetJdk = 1.7
```
Comment (rtyler) [inline build.gradle]: We should be able to put:

```groovy
targetJdk = targetCompatibility
```

here, that way there's only one set of defined constants

**Comment by rtyler** [comment]
@rkuris will you update the commit message with a link to where the PMD ruleset came from for posterity?

**Comment by rkuris** [comment]
@rtyler I just started with the full list and trimmed when I didn't agree with the rule for this project. The list of possible rules is here: http://pmd.sourceforge.net/pmd-5.1.1/rules/index.html

File: build.gradle (hunk 1 above)
Comment (rkuris) [inline build.gradle]: Fixed

**Pull request merged**